azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -1084,5 +1086,39 @@ func resourceLinuxVirtualMachineDelete(d *schema.ResourceData, meta interface{})
 		log.Printf("[DEBUG] Skipping Deleting OS Disk from Linux Virtual Machine %q (Resource Group %q)..", id.Name, id.ResourceGroup)
 	}
 
+	// Need to add a get and a state wait to avoid bug in network API where the attached disk(s) are not actually deleted
+	// Service team indicated that we need to do a get after VM delete call returns to verify that the VM and all attached
+	// disks have actually been deleted.
+
+	log.Printf("[INFO] verifying Linux Virtual Machine %q has been deleted", id.Name)
+	virtualMachine, _ := client.Get(ctx, id.ResourceGroup, id.Name, "")
```

Comment:
we should still check this error here, should network issues/an outage happen or something this could be an error

Reply:
Fixed.